cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternalsCommon.qll

```diff
@@ -152,6 +152,50 @@ predicate isWrite(Node0Impl value, Operand address, boolean certain) {
   )
 }
 
+predicate isAdditionalConversionFlow(Operand opFrom, Instruction instrTo) {
+  any(Indirection ind).isAdditionalConversionFlow(opFrom, instrTo)
+}
+
+predicate ignoreSourceVariableBase(Instruction base, Node0Impl value) {
+  any(Indirection ind).ignoreSourceVariableBase(base, value)
+}
+
+newtype TBaseSourceVariable =
```

Comment:
What exactly does `Base` refer to here?

Reply:
Mostly asking because we had two concepts a `BaseSourceVariable` and a `SourceVariable` and they're now starting to share things.

Reply:
`Base` because it doesn't count the number of indirections. This is done in the `SourceVariable` class in `SsaInternals.qll` in an IPA branch with two columns (a base source variable and an integer).

Reply:
Oh, yes. I see. `TSourceVariable` actually has two branches similar to `TBaseSourceVariable`, and it should really just have 1 branch containing a base and an integer column. That's definitely an easy fix to do.

Reply:
^ But let's save that for another PR.

Reply:
That I had not even spotted yet. I tripped over the special ssa0 case, where `SourceVariable` is effectively the same as `BaseSourceVariable` (I probably should have commented there). But, given our earlier discussion, that makes sense, I think.

What you seem to suggest seems a nice simplification.